nix/README.md

```diff
@@ -0,0 +1,36 @@
+# Nix
+
+This folder (along with the top level `flake.nix`) contains Nix configurations to enable this project to be built by `nix`. These configurations are mostly wrappers around Makefile goals. The goal of this work is to enable people who use `nix` to integrate this project more easily in their workflows. It is not the goal of this work to replace the Makefile, and any Nix configurations added should wrap Makefile goals whenever possible so binaries built with `nix` will be identical to those built without.
+
+## What is Nix
+
+Nix is a package manager that can aide in reproducible builds.
+
+## Using Nix
+
+Using `nix` with this repository is entirely optional. You are welcome to just use the `Makefile` and maintain versions of dependent binaries in your own way.
+
+Nix ships with commands such as `nix build` which will run our build (as defined in the `Makefile`) in a shell where the `$PATH` has been configured to have the specified version of all dependent binaries required to build and test our software (ie. `go`, `golangci-lint`, `jq`, `jsonnet`, etc), and put the resulting binary in a `result` folder. It also has a command `nix develop` which will dump you into a bash shell with a similarly configured `$PATH`. 
+
+So, for example, if the CI lint job fails, but it's passing locally, you could run the following to make sure it's not a problem with the version of the linter installed locally on you machine:
+
+From the root of the repo:
+
+```console
+nix develop
+make lint
+```
+
+To build the repo (including running tests), from the root of the repo you can run `nix build`.
+
+## Installing Nix
+
+Nix is supported on Linux, MacOS, and Windows (WSL2). Check [here](https://nixos.org/download.html#download-nix) for installation instructions for your specific platform.
+
+You will also need to enable the Flakes feature to use Nix with this repo. See this [wiki](https://nixos.wiki/wiki/Flakes) for instructions on enabling Flakes.
+
+## Dealing with .git
```

Comment:
Have you tried the approach lined out in: https://stackoverflow.com/questions/67025103/how-inject-git-revision-into-nix-derivation

Reply:
I think I am using the approach lined out in that link, which if I read correctly is to use `self.rev` for flakes? Maybe I'm misunderstanding if there's another workaround in that article?

Reply:
Ahh, I see, the suggestion is to use `(builtins.fetchGit args).rev`. That will not work, as the url here would be `./..` or `self`, which in either case points to the source in the nix store which has already been stripped of git info, so `fetchGit` will complain that this is not a git repo.